pexec/managed_process_test.go

```diff
@@ -193,6 +210,38 @@ func TestManagedProcessManage(t *testing.T) {
 			test.That(t, err.Error(), test.ShouldContainSubstring, "exit status 1")
 		}
 	})
+	t.Run("OnCrashHandler", func(t *testing.T) {
+		logger := golog.NewTestLogger(t)
+
+		crashHandlerCalledEnough := make(chan struct{})
+		var onCrashHandlerCallCount atomic.Uint64
+		proc := NewManagedProcess(ProcessConfig{
+			Name: "bash",
+			Args: []string{"-c", "exit 1"},
+			OnCrashHandler: func() bool {
+				// Close channel and return false (no restart) only after 5 restarts.
+				if onCrashHandlerCallCount.Add(1) == 5 {
+					close(crashHandlerCalledEnough)
+					return false
+				}
+				return true
+			},
+		}, logger)
+		test.That(t, proc.Start(context.Background()), test.ShouldBeNil)
+
+		<-crashHandlerCalledEnough
+
+		// Assert onCrashHandlerCallCount is exactly five even after waiting a few
+		// seconds (no further restarts).
+		time.Sleep(2 * time.Second)
```

Comment:
It seems like you should not need this sleep. by calling stop later and switching it to >= 5, you know test messed up if it double closes